Unable to restart BigchainDB and Tendermint at block height 1

## Description

* BigchainDB version: master - (https://github.com/bigchaindb/bigchaindb/commit/9bf09324df1c63b7fbef0b24830deb5bd80b4dae)
* Tendermint version: `0.22.8`
* MongoDB version: `3.6.3`
* OS: `Ubuntu 18.04.1 LTS bionic`

Unable to restart BigchainDB and Tendermint at block height `1` of Tendermint. I get the following errors in BigchainDB logs:

```
[2018-09-05 14:52:11] [ERROR] (bigchaindb.core) Got invalid InitChain ABCI request (time: 1536156417
chain_id: "test-chain-E0e9OT"
consensus_params {
  block_size {
    max_bytes: 22020096
    max_txs: 10000
    max_gas: -1
  }
  tx_size {
    max_bytes: 10240
    max_gas: -1
  }
  block_gossip {
    block_part_size_bytes: 65536
  }
}
validators {
  address: "r;J\001[\375\277\366-\247\242/5\246\362f\312b\214\342"
  pub_key {
    type: "ed25519"
    data: "U\017\277\306\340*\323\301\257%F\013\266\033\022\215\321\311\0160\363\007\276ow\343\323l\r4-\277"
  }
  power: 10
}
) - the chain {chain_id} is already synced. (MainProcess - pid: 6107)
``` 
and the system crashes.

### Steps to reproduce
 
- Install MongoDB
- Install Tendermint
- Install BigchainDB(master-(https://github.com/bigchaindb/bigchaindb/commit/9bf09324df1c63b7fbef0b24830deb5bd80b4dae))
- Run MongoDB + Tendermint + BigchainDB
- Kill Tendermint and BigchainDB
- Restart Tendermint and BigchainDB
``` bash
# automatically start mongodb as well with Ubuntu 18
$ sudo apt install mongodb 
$ curl -fOL https://github.com/tendermint/tendermint/releases/download/v0.22.8/tendermint_0.22.8_linux_amd64.zip
$ unzip unzip tendermint_0.22.8_linux_amd64.zip
$ mv tendermint /usr/local/bin/
$ git clone https://github.com/bigchaindb/bigchaindb.git
$ cd bigchaindb/
$ pip install .
$ tendermint init
$ tendermint node
$ bigchaindb start
$ pkill bigchaindb
$ pkill tendermint
$ tendermint node
$ bigchaindb start
```

### Workaround

- Drop database, reset tendermint
- Once the system is up, commit a transaction to the system and everything recovers normally.

--- a/bigchaindb/core.py
+++ b/bigchaindb/core.py
@@ -232,12 +232,13 @@ class App(BaseApplication):
         # register a new block only when new transactions are received
         if self.block_txn_ids:
             self.bigchaindb.store_bulk_transactions(self.block_transactions)
-            block = Block(app_hash=self.block_txn_hash,
-                          height=self.new_height,
-                          transactions=self.block_txn_ids)
-            # NOTE: storing the block should be the last operation during commit
-            # this effects crash recovery. Refer BEP#8 for details
-            self.bigchaindb.store_block(block._asdict())
+
+        block = Block(app_hash=self.block_txn_hash,
+                      height=self.new_height,
+                      transactions=self.block_txn_ids)
+        # NOTE: storing the block should be the last operation during commit
+        # this effects crash recovery. Refer BEP#8 for details
+        self.bigchaindb.store_block(block._asdict())
 
         logger.debug('Commit-ing new block with hash: apphash=%s ,'
                      'height=%s, txn ids=%s', data, self.new_height,

--- a/tests/tendermint/test_integration.py
+++ b/tests/tendermint/test_integration.py
@@ -101,7 +101,7 @@ def test_app(b, init_chain_request):
 
     block0 = b.get_latest_block()
     assert block0
-    assert block0['height'] == 1
+    assert block0['height'] == 2
 
     # when empty block is generated hash of previous block should be returned
     assert block0['app_hash'] == new_block_hash

